packages/medusa/src/api/utils/middlewares/products/variant-inventory-quantity.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  ContainerRegistrationKeys,
+  remoteQueryObjectFromString,
+} from "@medusajs/utils"
+import { MedusaRequest } from "../../../../types/routing"
+
+export async function getVariantInventoryItems({
+  req,
+  variantIds,
+  additionalFilters = {},
+  asMap = true,
+}) {
+  const remoteQuery = req.scope.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
+
+  const linkQuery = remoteQueryObjectFromString({
+    entryPoint: "product_variant_inventory_item",
```

Comment:
suggestion: for the link query, you can use `service: LINKS.ProductVariantInventoryItem`

Reply:
This approach should be the preferred one a bit like resolving from ModuleRegistrationName etc wdyt?

Reply:
Yes good push, let me add that in